### PR TITLE
Add product component

### PR DESCRIPTION
**Pull request by ShelOrock**
Now error handlers every way i can think of: 
1. name and price required.
2. price and inventory must be numbers.
3. File must be .png, .jpg or .gif.
4. File must be less than 5MB.

Paginated Routes added.

Acceptance criteria: 
run app and navigate to local host/api/users or products.
Add query params in url like so: http://localhost:3000/api/products?limit=5&page=0
Note: pages are 0-indexed.

Would love advice on how to break up the form: I think that component is waaay to complex. I'm thinking a dedicated error console, moving inputs into their own components, and moving methods into utils folder. Also toast appears only if button is successfully pressed. Not if post request is successful. Would love advice here too.

More forms coming 🤪🤪🤪

### CODE REVIEW

**Review by nries1 — DISMISSED** [review]
@ShelOrock. This passes the acceptance criteria. I'll take a look at the form and see what advice I can give for simplifying it. Hopefully we can discuss tomorrow in class. At some point can you take a look at the pagination get route in server/api/products? I think this may log an exception in node because it sends res to the client without returning out of the route.

```
router.get('/', paginate(Product), (req, res, next) => {
  res.send(res.foundModels).catch(e => {
    res.status(404);
    next(e);
  });
});
```

**Comment by ShelOrock** [comment]
error in node fixed!

File: server/api/utils.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //Pagination middleware. 
 const paginate = (model) => {
+	debugger;
```
Comment (nries1) [inline server/api/utils.js]: still need debugger?

**Review by nries1 — DISMISSED** [review]
Very nice!

**Review by raymondng1 — APPROVED** [review]
Nice !

**Pull request merged**